src/groupByKey.ts

```diff
@@ -2,6 +2,40 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: [], key: string): GroupsMap<typeof items> {
+  const result: GroupsMap<typeof items> = {};
+
+  items.map((item) => {
+    result[item[key]] = [];
+
+    return result;
+  });
+
+  items.map((item) => {
```

Comment:
You don't need to duplicate the same method for the array twice, you need to write the logic inside one method, think how it can be fixed